### ISSUE TITLE
Add Layout/SpaceBeforeBrackets method exclusion

### ISSUE THREAD

**Issue by bkuhlmann** [rubocop-hq/rubocop#9307]
## Is your feature request related to a problem? Please describe.

It would be nice if we could configure this cop to exclude methods, especially custom conversion functions, that accept arrays without being flagged as a violation. Example:

``` 
spec/lib/versionaire/cast_spec.rb:58:32: C: [Correctable] Layout/SpaceBeforeBrackets: Remove the space before the opening brackets. (https://rubystyle.guide#space-in-brackets-access)
        result = proc { Version [1, 2, 3, 4] }
```

This is happening because the Versionaire gem refines `Kernel` to allow one to cast an array (and other primitives) to a version type (see highlighted Line 6 below):

![image](https://user-images.githubusercontent.com/50245/103305419-54355400-49c8-11eb-9913-09b66e930ba5.png)

## Describe the solution you'd like

It'd be great if I could configure `Layout/SpaceBeforeBrackets` to exclude the `Version` function.

## Describe alternatives you've considered

For the time being, I have to exclude files instead. Example:

``` yaml
Layout/SpaceBeforeBrackets:
  Exclude:
    - "path/to/file/using/version/cast.rb"
```

## Additional context

``` 
rubocop -v
warning: parser/current is loading parser/ruby30, which recognizes
warning: 3.0.0-dev-compliant syntax, but you are running 3.0.0.
warning: please see https://github.com/whitequark/parser#compatibility-with-ruby-mri.
1.7.0
```

**Comment by koic** [comment]
#9291 no longer detects this and configuration will not be required. Please wait for the next release. Thank you for your feedback.
